src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -0,0 +1,652 @@
+    @unpack derivative_dhat = dg.basis
+    @threaded for element in eachelement(dg, cache)
+
+      # Calculate volume terms in one element
+      for j in eachnode(dg), i in eachnode(dg)
+        u_node = get_node_vars(u, equations_parabolic, dg, i, j, element)
+
+        for ii in eachnode(dg)
+          multiply_add_to_node_vars!(u_grad[1], derivative_dhat[ii, i], u_node, equations_parabolic, dg, ii, j, element)
+        end
+
+        for jj in eachnode(dg)
+          multiply_add_to_node_vars!(u_grad[2], derivative_dhat[jj, j], u_node, equations_parabolic, dg, i, jj, element)
+        end
+      end
+    end
+  end
+
+  # Prolong solution to interfaces
+  @trixi_timeit timer() "prolong2interfaces" begin
+    @unpack interfaces = cache_parabolic
+    @unpack orientations = interfaces
+
+    @threaded for interface in eachinterface(dg, cache)
+      left_element  = interfaces.neighbor_ids[1, interface]
+      right_element = interfaces.neighbor_ids[2, interface]
+
+      if orientations[interface] == 1
+        # interface in x-direction
+        for j in eachnode(dg), v in eachvariable(equations_parabolic)
+          interfaces.u[1, v, j, interface] = u[v, nnodes(dg), j, left_element]
+          interfaces.u[2, v, j, interface] = u[v,          1, j, right_element]
+        end
+      else # if orientations[interface] == 2
+        # interface in y-direction
+        for i in eachnode(dg), v in eachvariable(equations_parabolic)
+          interfaces.u[1, v, i, interface] = u[v, i, nnodes(dg), left_element]
+          interfaces.u[2, v, i, interface] = u[v, i,          1, right_element]
+        end
+      end
+    end
+  end
+
+  # Calculate interface fluxes
+  @trixi_timeit timer() "interface flux" begin
+    @unpack surface_flux_values = cache_parabolic.elements
+    @unpack neighbor_ids, orientations = cache_parabolic.interfaces
+
+    @threaded for interface in eachinterface(dg, cache_parabolic)
+      # Get neighboring elements
+      left_id  = neighbor_ids[1, interface]
+      right_id = neighbor_ids[2, interface]
+
+      # Determine interface direction with respect to elements:
+      # orientation = 1: left -> 2, right -> 1
+      # orientation = 2: left -> 4, right -> 3
+      left_direction  = 2 * orientations[interface]
+      right_direction = 2 * orientations[interface] - 1
+
+      for i in eachnode(dg)
+        # Call pointwise Riemann solver
+        u_ll, u_rr = get_surface_node_vars(cache_parabolic.interfaces.u,
+                                           equations_parabolic, dg, i, interface)
+        flux = 0.5 * (u_ll + u_rr)
+
+        # Copy flux to left and right element storage
+        for v in eachvariable(equations_parabolic)
+          surface_flux_values[v, i, left_direction,  left_id]  = flux[v]
+          surface_flux_values[v, i, right_direction, right_id] = flux[v]
+        end
+      end
+    end
+  end
+
+  # Prolong solution to boundaries
+  @trixi_timeit timer() "prolong2boundaries" begin
```

Comment:
Use `prolong2boundaries!` instead?

Reply:
Done in f5f48fc04e822340e8613f62c2558c280beaafd3